From 22d3f3c43b1a0a4165fb66a3db6ac1bab45311b5 Mon Sep 17 00:00:00 2001
From: koebi <koebischnell@…>
Date: Sat, 7 Sep 2013 19:41:16 +0200
Subject: [PATCH] Run authentification in different process

This fixes #895
---
 i3lock.c | 83 +++++++++++++++++++++++++++++++++++++++++-----------------------
 1 file changed, 53 insertions(+), 30 deletions(-)

diff --git a/i3lock.c b/i3lock.c
index 74e15f2..c845a6a 100644
--- a/i3lock.c
+++ b/i3lock.c
@@ -22,6 +22,7 @@
 #include <string.h>
 #include <ev.h>
 #include <sys/mman.h>
+#include <sys/wait.h>
 #include <X11/XKBlib.h>
 #include <X11/extensions/XKBfile.h>
 #include <xkbcommon/xkbcommon.h>
@@ -198,6 +199,46 @@ static void clear_input(void) {
     redraw_screen();
     unlock_state = STATE_KEY_PRESSED;
 }
+static void child_cb(EV_P_ ev_child *child_watcher, int revents) {
+    if (child_watcher->rstatus == PAM_SUCCESS){
+        DEBUG("Authentication successfull\n");
+        clear_password_memory();
+
+        ev_child_stop(main_loop, child_watcher);
+        free(child_watcher);
+        
+        exit(0);
+    } else {
+
+        if (debug_mode)
+            fprintf(stderr, "Authentication failure\n");
+        
+        ev_child_stop(main_loop, child_watcher);
+        free(child_watcher);
+
+        pam_state = STATE_PAM_WRONG;
+        clear_input();
+        redraw_screen();
+
+        /* Clear this state after 2 seconds (unless the user enters another
+         * password during that time). */
+        ev_now_update(main_loop);
+        if ((clear_pam_wrong_timeout = calloc(sizeof(struct ev_timer), 1))) {
+            ev_timer_init(clear_pam_wrong_timeout, clear_pam_wrong, 2.0, 0.);
+            ev_timer_start(main_loop, clear_pam_wrong_timeout);
+        }
+
+        /* Cancel the clear_indicator_timeout, it would hide the unlock indicator
+         * too early. */
+        stop_clear_indicator_timeout();
+
+        /* beep on authentication failure, if enabled */
+        if (beep) {
+            xcb_bell(conn, 100);
+            xcb_flush(conn);
+        }
+    }
+}
 
 static void input_done(void) {
     if (clear_pam_wrong_timeout) {
@@ -208,36 +249,18 @@ static void input_done(void) {
 
     pam_state = STATE_PAM_VERIFY;
     redraw_screen();
-
-    if (pam_authenticate(pam_handle, 0) == PAM_SUCCESS) {
-        DEBUG("successfully authenticated\n");
-        clear_password_memory();
-        exit(0);
-    }
-
-    if (debug_mode)
-        fprintf(stderr, "Authentication failure\n");
-
-    pam_state = STATE_PAM_WRONG;
-    clear_input();
-    redraw_screen();
-
-    /* Clear this state after 2 seconds (unless the user enters another
-     * password during that time). */
-    ev_now_update(main_loop);
-    if ((clear_pam_wrong_timeout = calloc(sizeof(struct ev_timer), 1))) {
-        ev_timer_init(clear_pam_wrong_timeout, clear_pam_wrong, 2.0, 0.);
-        ev_timer_start(main_loop, clear_pam_wrong_timeout);
-    }
-
-    /* Cancel the clear_indicator_timeout, it would hide the unlock indicator
-     * too early. */
-    stop_clear_indicator_timeout();
-
-    /* beep on authentication failure, if enabled */
-    if (beep) {
-        xcb_bell(conn, 100);
-        xcb_flush(conn);
+    
+    /* fork to unblock pam_authenticate */
+    pid_t cpid;
+    cpid = fork();
+    struct ev_child *child_watcher = calloc(sizeof(struct ev_io), 1);
+    ev_child_init(child_watcher, child_cb, cpid, 0);
+    ev_child_set(child_watcher, cpid, 0);
+    ev_child_start (EV_DEFAULT_ child_watcher);
+    if (cpid == 0) {    
+        exit(pam_authenticate(pam_handle, 0));
+    } else if (cpid < 0) {
+        err(1, "Could not fork");
     }
 }
 
-- 
1.8.4.rc3

