From b1ff96942ae51aeb2479b0a57a6438498e955d32 Mon Sep 17 00:00:00 2001
From: Axel Wagner <mail@…>
Date: Tue, 9 Apr 2013 18:31:25 +0200
Subject: [PATCH] Set window name

---
 xcb.c |   11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/xcb.c b/xcb.c
index 6498241..4e88666 100644
--- a/xcb.c
+++ b/xcb.c
@@ -9,6 +9,7 @@
  */
 #include <xcb/xcb.h>
 #include <xcb/xcb_image.h>
+#include <xcb/xcb_atom.h>
 #include <xcb/dpms.h>
 #include <stdio.h>
 #include <stdlib.h>
@@ -133,6 +134,16 @@ xcb_window_t open_fullscreen_window(xcb_connection_t *conn, xcb_screen_t *scr, c
                       mask,
                       values);
 
+    char *name = "i3lock";
+    xcb_change_property(conn,
+                        XCB_PROP_MODE_REPLACE,
+                        win,
+                        XCB_ATOM_WM_NAME,
+                        XCB_ATOM_STRING,
+                        8,
+                        strlen (name),
+                        name);
+
     /* Map the window (= make it visible) */
     xcb_map_window(conn, win);
 
-- 
1.7.10.4

