From e6bc39865ea6c53df64a0da3405c28e92c39ec12 Mon Sep 17 00:00:00 2001
From: Arnaud Degroote <degroote@…>
Date: Sun, 6 Oct 2013 18:13:29 +0000
Subject: [PATCH 3/3] Allow to print cpu usage on NetBSD

---
 src/print_cpu_usage.c | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/src/print_cpu_usage.c b/src/print_cpu_usage.c
index ecc81a2..80944c5 100644
--- a/src/print_cpu_usage.c
+++ b/src/print_cpu_usage.c
@@ -20,6 +20,13 @@
 #include <sys/resource.h>
 #endif
 
+#if defined(__NetBSD__)
+#include <sys/param.h>
+#include <sys/resource.h>
+#include <sys/sysctl.h>
+#include <sys/sched.h>
+#endif
+
 #include "i3status.h"
 
 static int prev_total = 0;
@@ -50,9 +57,9 @@ void print_cpu_usage(yajl_gen json_gen, char *buffer, const char *format) {
         diff_usage = (diff_total ? (1000 * (diff_total - diff_idle)/diff_total + 5)/10 : 0);
         prev_total = curr_total;
         prev_idle  = curr_idle;
-#elif defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__DragonFly__)
+#elif defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__DragonFly__)
 
-#if defined(__FreeBSD__) || defined(__DragonFly__)
+#if defined(__FreeBSD__) || defined(__DragonFly__) || defined(__NetBSD__)
         size_t size;
         long cp_time[CPUSTATES];
         size = sizeof cp_time;
-- 
1.8.4

