From 2dae41fbfc34bbe2f2d725b0ca8716bc5a97f77e Mon Sep 17 00:00:00 2001
From: Baptiste Daroussin <bapt@…>
Date: Tue, 19 Mar 2013 18:32:08 +0100
Subject: [PATCH 2/2] Support mixer_idx with OSS

---
 src/print_volume.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/print_volume.c b/src/print_volume.c
index 1b03e5e..d53de06 100644
--- a/src/print_volume.c
+++ b/src/print_volume.c
@@ -175,12 +175,16 @@ void print_volume(yajl_gen json_gen, char *buffer, const char *fmt, const char *
 	}
 #endif
 #if defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__DragonFly__)
-        char mixerpath[] = "/dev/mixer";
+        char *mixerpath;
         int mixfd, vol, devmask = 0;
         pbval = 1;
 
+        asprintf(&mixerpath, "/dev/mixer%d", mixer_idx);
+
         if ((mixfd = open(mixerpath, O_RDWR)) < 0)
                 return;
+
+        free(mixerpath);
         if (ioctl(mixfd, SOUND_MIXER_READ_DEVMASK, &devmask) == -1)
                 return;
         if (ioctl(mixfd, MIXER_READ(0),&vol) == -1)
-- 
1.8.1.3

