From 12fa2e3de2412e1cf96ae81dad4a63e1e6141607 Mon Sep 17 00:00:00 2001
From: Marius Muja <mariusm@…>
Date: Mon, 31 Dec 2012 03:04:53 -0800
Subject: [PATCH] Fix decoration rect size for windows without border

---
 src/render.c |   24 ++++++++++++++++--------
 1 file changed, 16 insertions(+), 8 deletions(-)

diff --git a/src/render.c b/src/render.c
index 476d1ca..4dcc981 100644
--- a/src/render.c
+++ b/src/render.c
@@ -298,16 +298,24 @@ void render_con(Con *con, bool render_fullscreen) {
             }
 
             /* first we have the decoration, if this is a leaf node */
-            if (con_is_leaf(child) && child->border_style == BS_NORMAL) {
-                /* TODO: make a function for relative coords? */
-                child->deco_rect.x = child->rect.x - con->rect.x;
-                child->deco_rect.y = child->rect.y - con->rect.y;
+            if (con_is_leaf(child)) {
+                if (child->border_style == BS_NORMAL) {
+                    /* TODO: make a function for relative coords? */
+                    child->deco_rect.x = child->rect.x - con->rect.x;
+                    child->deco_rect.y = child->rect.y - con->rect.y;
 
-                child->rect.y += deco_height;
-                child->rect.height -= deco_height;
+                    child->rect.y += deco_height;
+                    child->rect.height -= deco_height;
 
-                child->deco_rect.width = child->rect.width;
-                child->deco_rect.height = deco_height;
+                    child->deco_rect.width = child->rect.width;
+                    child->deco_rect.height = deco_height;
+                }
+                else {
+                    child->deco_rect.x = 0;
+                    child->deco_rect.y = 0;
+                    child->deco_rect.width = 0;
+                    child->deco_rect.height = 0;
+                }
             }
         }
 
-- 
1.7.10.4

