From a7606d89c5a0179bad39bbd8a4876cf2a6e5f96f Mon Sep 17 00:00:00 2001
From: Emil Mikulic <emikulic@…>
Date: Sun, 13 Jan 2013 23:23:43 +1100
Subject: [PATCH] Make sure colorful_output variable is initialized.

---
 src/print_cpu_temperature.c |   10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/print_cpu_temperature.c b/src/print_cpu_temperature.c
index 96fa4b7..e25cea5 100644
--- a/src/print_cpu_temperature.c
+++ b/src/print_cpu_temperature.c
@@ -38,7 +38,7 @@ void print_cpu_temperature_info(yajl_gen json_gen, char *buffer, int zone, const
 #ifdef THERMAL_ZONE
         const char *walk;
         char *outwalk = buffer;
-	bool colorful_output;
+        bool colorful_output = false;
 
         if (path == NULL)
                 asprintf(&thermal_zone, THERMAL_ZONE, zone);
@@ -69,8 +69,10 @@ void print_cpu_temperature_info(yajl_gen json_gen, char *buffer, int zone, const
                                         colorful_output = true;
                                 }
                                 outwalk += sprintf(outwalk, "%ld", (temp/1000));
-                                if (colorful_output)
+                                if (colorful_output) {
                                         END_COLOR;
+                                        colorful_output = false;
+                                }
                         }
 #elif defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__DragonFly__)
                         int sysctl_rslt;
@@ -116,8 +118,10 @@ void print_cpu_temperature_info(yajl_gen json_gen, char *buffer, int zone, const
 
 				outwalk += sprintf(outwalk, "%.2f", MUKTOC(sensor.value));
 
-				if (colorful_output)
+				if (colorful_output) {
 					END_COLOR;
+                                        colorful_output = false;
+                                }
 			}
 		}
 	}
-- 
1.7.10.4

