From 1f1e7c12a03f96cd81d3176fb47a92f1be1ee910 Mon Sep 17 00:00:00 2001
From: Axel Wagner <mail@…>
Date: Tue, 9 Apr 2013 18:31:25 +0200
Subject: [PATCH] Set window name

---
 xcb.c |   12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/xcb.c b/xcb.c
index 6498241..5f3d298 100644
--- a/xcb.c
+++ b/xcb.c
@@ -9,10 +9,12 @@
  */
 #include <xcb/xcb.h>
 #include <xcb/xcb_image.h>
+#include <xcb/xcb_atom.h>
 #include <xcb/dpms.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <stdbool.h>
+#include <string.h>
 #include <unistd.h>
 #include <assert.h>
 #include <err.h>
@@ -133,6 +135,16 @@ xcb_window_t open_fullscreen_window(xcb_connection_t *conn, xcb_screen_t *scr, c
                       mask,
                       values);
 
+    char *name = "i3lock";
+    xcb_change_property(conn,
+                        XCB_PROP_MODE_REPLACE,
+                        win,
+                        XCB_ATOM_WM_NAME,
+                        XCB_ATOM_STRING,
+                        8,
+                        strlen(name),
+                        name);
+
     /* Map the window (= make it visible) */
     xcb_map_window(conn, win);
 
-- 
1.7.10.4

