From 58b1f2d1292c7165c0e17722f3a75b2df5592c8f Mon Sep 17 00:00:00 2001
From: Baptiste Daroussin <bapt@…>
Date: Wed, 20 Mar 2013 16:59:55 +0100
Subject: [PATCH 2/3] shm_unlink the created/open shm

---
 src/log.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/log.c b/src/log.c
index e858f94..e490cf3 100644
--- a/src/log.c
+++ b/src/log.c
@@ -116,7 +116,7 @@ void init_logging(void) {
 
         if (ftruncate(logbuffer_shm, logbuffer_size) == -1) {
             close(logbuffer_shm);
-            shm_unlink("/i3-log-");
+            shm_unlink(shmlogname);
             fprintf(stderr, "Could not ftruncate SHM segment for the i3 log: %s\n", strerror(errno));
             return;
         }
@@ -124,7 +124,7 @@ void init_logging(void) {
         logbuffer = mmap(NULL, logbuffer_size, PROT_READ | PROT_WRITE, MAP_SHARED, logbuffer_shm, 0);
         if (logbuffer == MAP_FAILED) {
             close(logbuffer_shm);
-            shm_unlink("/i3-log-");
+            shm_unlink(shmlogname);
             fprintf(stderr, "Could not mmap SHM segment for the i3 log: %s\n", strerror(errno));
             logbuffer = NULL;
             return;
-- 
1.8.1.3

