From d932f5bbcf37520583b1f10cf53fe3666dc3301a Mon Sep 17 00:00:00 2001
From: Tony Crisci <tony@…>
Date: Mon, 10 Mar 2014 04:29:44 -0400
Subject: [PATCH 4/4] Free window ran_assignments on tree_close

---
 src/tree.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/tree.c b/src/tree.c
index e3a5495..80af522 100644
--- a/src/tree.c
+++ b/src/tree.c
@@ -260,6 +260,7 @@ bool tree_close(Con *con, kill_window_t kill_window, bool dont_kill_parent, bool
         FREE(con->window->class_class);
         FREE(con->window->class_instance);
         i3string_free(con->window->name);
+        FREE(con->window->ran_assignments);
         FREE(con->window);
     }
 
-- 
1.9.0

