From b284b35e82bc74dd78d6713adce8f294993dd81d Mon Sep 17 00:00:00 2001
From: TheKK <thumbd03803@…>
Date: Sun, 16 Mar 2014 21:54:42 +0800
Subject: [PATCH] Replace '1' with "EXIT_FAILURE"

---
 i3lock.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/i3lock.c b/i3lock.c
index 1599282..60a3330 100644
--- a/i3lock.c
+++ b/i3lock.c
@@ -664,7 +664,7 @@ int main(int argc, char *argv[]) {
     };
 
     if ((username = getenv("USER")) == NULL)
-        errx(1, "USER environment variable not set, please set it.\n");
+        errx(EXIT_FAILURE, "USER environment variable not set, please set it.\n");
 
     while ((o = getopt_long(argc, argv, "hvnbdc:p:ui:te", longopts, &optind)) != -1) {
         switch (o) {
@@ -687,7 +687,7 @@ int main(int argc, char *argv[]) {
                 arg++;
 
             if (strlen(arg) != 6 || sscanf(arg, "%06[0-9a-fA-F]", color) != 1)
-                errx(1, "color is invalid, it must be given in 3-byte hexadecimal format: rrggbb\n");
+                errx(EXIT_FAILURE, "color is invalid, it must be given in 3-byte hexadecimal format: rrggbb\n");
 
             break;
         }
@@ -706,7 +706,7 @@ int main(int argc, char *argv[]) {
             } else if (!strcmp(optarg, "default")) {
                 curs_choice = CURS_DEFAULT;
             } else {
-                errx(1, "i3lock: Invalid pointer type given. Expected one of \"win\" or \"default\".\n");
+                errx(EXIT_FAILURE, "i3lock: Invalid pointer type given. Expected one of \"win\" or \"default\".\n");
             }
             break;
         case 'e':
@@ -717,7 +717,7 @@ int main(int argc, char *argv[]) {
                 debug_mode = true;
             break;
         default:
-            errx(1, "Syntax: i3lock [-v] [-n] [-b] [-d] [-c color] [-u] [-p win|default]"
+            errx(EXIT_FAILURE, "Syntax: i3lock [-v] [-n] [-b] [-d] [-c color] [-u] [-p win|default]"
             " [-i image.png] [-t] [-e]"
             );
         }
-- 
1.8.3.2

