From 0c925e9e91670d1e26fcefe69d06d02bc83a3f3a Mon Sep 17 00:00:00 2001
From: Alexander Berntsen <alexander@…>
Date: Wed, 5 Jun 2013 14:59:05 +0200
Subject: [PATCH 1/4] Make default shmlog_size a #define, SHMLOG_SIZE

---
 include/shmlog.h | 2 ++
 src/main.c       | 4 +++-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/include/shmlog.h b/include/shmlog.h
index fd3f53e..f7a259f 100644
--- a/include/shmlog.h
+++ b/include/shmlog.h
@@ -14,6 +14,8 @@
 #include <stdint.h>
 #include <pthread.h>
 
+#define SHMLOG_SIZE 25 * 1024 * 1024
+
 /*
  * Header of the shmlog file. Used by i3/src/log.c and i3/i3-dump-log/main.c.
  *
diff --git a/src/main.c b/src/main.c
index db3aca4..def5247 100644
--- a/src/main.c
+++ b/src/main.c
@@ -19,6 +19,8 @@
 #include <sys/mman.h>
 #include <sys/stat.h>
 #include "all.h"
+/* SHMLOG_SIZE */
+#include "shmlog.h"
 
 #include "sd-daemon.h"
 
@@ -291,7 +293,7 @@ int main(int argc, char *argv[]) {
     init_logging();
 
     /* On non-release builds, disable SHM logging by default. */
-    shmlog_size = (is_debug_build() ? 25 * 1024 * 1024 : 0);
+    shmlog_size = (is_debug_build() ? SHMLOG_SIZE : 0);
 
     start_argv = argv;
 
-- 
1.8.1.5

